### PR TITLE
Support user-configurable display parameters

### PR DESCRIPTION
**Pull request by oskarpearson**
Allow users to select either mg/dL or mmol/L display parameters, so as to
support other countries.
- Rename ReportFile class to Settings, so that it's more general

### CODE REVIEW

**Comment by loudnate** [comment]
Great work! I'm looking forward to merging this when I'm next on my laptop.

There is no versioning strategy for this project yet, so for now I'm going to continue to accept PRs straight to master.

File: chart.py [hunk 1]
```diff
@@ -1,24 +1,34 @@
 # coding=utf-8
 from dateutil.parser import parse
 
+# How to convert from mg/dL (US format) to mmol/L (non-US format)
+MMOLL_CONVERT_FACTOR = 1.0 / 18.0
```
Comment (loudnate) [inline chart.py:5]: Some minor feedback, but I think it'd be best to move the conversion higher up in the execution path. This file is really just munging for the GChart API.

Have you found a way to pull the unit preference from pump settings?

Comment (oskarpearson) [inline chart.py:5]: I'm trying to keep the conversion at the "view"/"presentation" layer (thinking of this as a semi-traditional model-view-controller type structure).

It's kinda like storing 'cents' in the DB, and then having code to conver them to "$1.99" in the view code.

I think that the approach OpenAPS has taken is a sensible one (just using one unit of measurement). I'm trying to preserve the sanctity of that, and do the conversion as close to the edge as possible.

EG - if we add code to the system that highlights something in red if sugars are high, the code there won't need a "if unit = mmol/L, then mark_high when greater than X, but if it's unit mg/dL then the high mark is Z"

It looks like I could get the display units with this. It'd require a patch to add the functionality to a report in openaps though.
https://github.com/bewest/decoding-carelink/blob/cff4aea826791244a14727e0e796a1216508718a/decocare/commands.py#L911

**Comment by loudnate** [comment]
Yeah, you make a good point that there are probably many more cases where configurable values should stay in one unit.

It's not really OpenAPS's decision to keep values in one unit, it's really because that's what the MM data format uses.

I'm good with this for now. Let's shoot for reading the units directly from the medtronic vendor report once that's available (PR's on openaps are welcome!)

**Pull request merged**